### PR TITLE
[BUG] Resolve `LSTMFCNClassifier` and `LSTMFCNRegressor` changing `callback` parameter 

### PR DESCRIPTION
**Pull request by ArthrowAbstract**
#### Reference Issues/PRs

Fixes check_estimator for #6153 and partially solves it.
#### What does this implement/fix? Explain your changes.

It corrects the `test_fit_does_not_overwrite_hyper_params` by replacing the `callback = []` to `callback = None` as originally referenced.
image

| Before Fix  | After Fix |
| ------------- | ------------- |
| image  |image |
| image | image  |

Furthermore, to make sure that other code wasn't compromised I ran 2 more `check_estimator`
| Before Fix  | After Fix |
| ------------- | ------------- |
| image | image |
|image| image|

#### Does your contribution introduce a new dependency? If yes, which one?
NA

##### For all contributions
- [x] I've added myself to the [list of contributors](https://github.com/sktime/sktime/blob/main/CONTRIBUTORS.md) with any new badges I've earned :-)
  How to: add yourself to the [all-contributors file](https://github.com/sktime/sktime/blob/main/.all-contributorsrc) in the `sktime` root directory (not the `CONTRIBUTORS.md`). Common badges: `code` - fixing a bug, or adding code logic. `doc` - writing or improving documentation or docstrings. `bug` - reporting or diagnosing a bug (get this plus `code` if you also fixed the bug in the PR).`maintenance` - CI, test framework, release.
  See here for [full badge reference](https://allcontributors.org/docs/en/emoji-key)
- [ ] Optionally, for added estimators: I've added myself and possibly to the `maintainers` tag - do this if you want to become the owner or maintainer of an estimator you added.
  See here for further details on the [algorithm maintainer role](https://www.sktime.net/en/latest/get_involved/governance.html#algorithm-maintainers).
- [x] The PR title starts with either [ENH], [MNT], [DOC], or [BUG]. [BUG] - bugfix, [MNT] - CI, test framework, [ENH] - adding or improving code, [DOC] - writing or improving documentation or docstrings.

##### For new estimators
- [ ] I've added the estimator to the API reference - in `docs/source/api_reference/taskname.rst`, follow the pattern.
- [ ] I've added one or more illustrative usage examples to the docstring, in a pydocstyle compliant `Examples` section.
- [ ] If the estimator relies on a soft dependency, I've set the `python_dependencies` tag and ensured
  dependency isolation, see the [estimator dependencies guide](https://www.sktime.net/en/latest/developer_guide/dependencies.html#adding-a-soft-dependency).

### CODE REVIEW

**Comment by fkiraly** [comment]
Thanks!

The pickling bug is common to all tensorflow based estimators, and tracked separately.

I would recommend:
add the `LSTFMFCNNRegressor` to `EXCLUDED_TESTS` in `tests._config` with the remaining failing test, but remove it from `EXCLUDED_ESTIMATORS`, so the tests run on the CI

Further, I think there are some code formatting errors, see here how to set linting up locally:
https://www.sktime.net/en/stable/developer_guide/coding_standards.html

**Comment by ArthrowAbstract** [comment]
@fkiraly I have updated the linting and `test._config` file.

**Review by fkiraly — CHANGES_REQUESTED** [review]
Ok, the tests are running now.

For the fix: please don't change the tests, these are there to ensure all estimators comply.

Please change only the estimator, that's where you need to apply the fix.

**Comment by ArthrowAbstract** [comment]
> Ok, the tests are running now.
> 
> For the fix: please don't change the tests, these are there to ensure all estimators comply.
> 
> Please change only the estimator, that's where you need to apply the fix.

Understood! I will update the files!

File: sktime/classification/deep_learning/lstmfcn.py [hunk 1]
```diff
@@ -132,8 +132,9 @@ def build_model(self, input_shape, n_classes, **kwargs):
             optimizer="adam",
             metrics=["accuracy"],
         )
-
-        self.callbacks = self.callbacks or []
+        # .get_params() returns an empty list for callback.
+        # inconsistent with function initial run where callbacks was set to None
+        self.callbacks = self.callbacks or None
```
Comment (fkiraly) [inline sktime/classification/deep_learning/lstmfcn.py:137]: `self.callbacks` should never be changed

Comment (ArthrowAbstract) [reply sktime/classification/deep_learning/lstmfcn.py:137]: If it is left to an empty list, then the error re-occur's.

Comment (fkiraly) [reply sktime/classification/deep_learning/lstmfcn.py:137]: yes, please

Comment (ArthrowAbstract) [reply sktime/classification/deep_learning/lstmfcn.py:137]: Got it! I will make correct changes asap!

**Comment by fkiraly** [comment]
I think there are more issues here than just the parameter - it appears the network does not build.

**Pull request merged**